lib/cosmos/tools/config_editor/system_config_dialog.rb

```diff
@@ -0,0 +1,158 @@
+      description.setWordWrap(true)
+      layout.addWidget(description)
+
+      select = Qt::Label.new("Select an existing system.txt file to base the new configuration on:")
+      select.setWordWrap(true)
+      layout.addWidget(select)
+      @system_combo = Qt::ComboBox.new
+      Dir[File.join(::Cosmos::USERPATH, 'config', 'system', '*.txt')].each do |system|
+        @system_combo.addItem(File.basename(system))
+      end
+      layout.addWidget(@system_combo)
+      layout.addSpacing(10)
+  
+      name_label = Qt::Label.new("Enter a name for the new system configuration that is descriptive "\
+        " but relatively short. For example, 'EMI', 'SW Test', etc:")
+      name_label.setWordWrap(true)
+      layout.addWidget(name_label)
+      @system_name = Qt::LineEdit.new
+      layout.addWidget(@system_name)
+      layout.addSpacing(10)
+
+      info = Qt::Label.new("This action will create a new COSMOS system.txt, "\
+        "cmd_tlm_server.txt, launcher.txt, and Windows Batch file appended with the specified name. "\
+        "For example, system_emi.txt, system_sw_test.txt, launcher_emi.txt, launcher_sw_test.txt, etc.")
+      info.setWordWrap(true)
+      layout.addWidget(info)
+
+      ok_button = Qt::PushButton.new('Ok')
+      connect(ok_button, SIGNAL('clicked()'), self, SLOT('accept()'))
+      cancel_button = Qt::PushButton.new('Cancel')
+      connect(cancel_button, SIGNAL('clicked()'), self, SLOT('reject()'))
+
+      hlayout = Qt::HBoxLayout.new
+      hlayout.addWidget(ok_button, 0, Qt::AlignLeft)
+      hlayout.addWidget(cancel_button, 0, Qt::AlignRight)
+      layout.addLayout(hlayout)
+
+      resize(500, 300)
+
+      self.show()
+      self.raise()
+      if self.exec() == Qt::Dialog::Accepted
+        build_system_config()
+      end
+      self.dispose()
+    end
+
+    def build_system_config
+      new_suffix = @system_name.text.downcase.gsub(' ','_')
+      existing_system = @system_combo.text
+      # Look for system_xxx.txt where we find the 'xxx' as the existing suffix
+      if existing = existing_system.scan(/.*?_(.*)\.txt/)[0]
+        existing_suffix = existing[0]
+      else
+        existing_suffix = nil
+      end
+
+      # First determine all the new files and make sure they don't already exist
+      new_system = File.join(Cosmos::USERPATH, 'config', 'system', "system_#{new_suffix}.txt")
+      return if file_exist?(new_system)
+      cmd_tlm_server_path = File.join('config', 'tools', 'cmd_tlm_server')
+      new_cmd_tlm_server = File.join(Cosmos::USERPATH, cmd_tlm_server_path, "cmd_tlm_server_#{new_suffix}.txt")
+      return if file_exist?(new_cmd_tlm_server)
+      launcher_path = File.join('config', 'tools', 'launcher')
+      new_launcher = File.join(Cosmos::USERPATH, launcher_path, "launcher_#{new_suffix}.txt")
+      return if file_exist?(new_launcher)
+      new_batch = File.join(Cosmos::USERPATH, "Launcher#{@system_name.text.gsub(' ','')}.bat")
+      return if file_exist?(new_batch)
```

Comment:
Shouldn't you alert the user of preexistings files rather than just returning?

Reply:
The file_exist? method does that.